bug: manager-api -p doesn't work

# Bug report

## Describe the bug

manager-api `-p` parameter doesn't work, it will prompt  `panic: fail to read configuration: ./conf/conf.yaml`

1. move `output` to another dir
2. use `ln -s` create manager-api soft links to `/usr/bin`
3. use manager-api -p `/path/to/output`
4. it will prompt  `panic: fail to read configuration: ./conf/conf.yaml`

@nic-chen

+1 for same issue.

After digging deeper into this issue, I found that `conf.WorkDir` would always be `.`, cause the config initialization process defined in the `init()` function under `github.com/apisix/manager-api/internal/log` package would be forever executed in prior to setting Cobra command flags. 

So I could get through with this via commenting out https://github.com/apache/apisix-dashboard/blob/62d1c439f37c98ca459f1ce2a0c15273575d2bce/api/internal/conf/conf.go#L103-L106 .

While if proceed, I would still got the following error:

```
➜  apache-apisix ./manager-api -p output
panic: couldn't open sink "logs/error.log": open logs/error.log: no such file or directory
goroutine 1 [running]:
github.com/apisix/manager-api/internal/log.fileWriter(0x1010200, 0x1650e252fc378834, 0xc00039fdf0)
	/Users/imjoey/Work/apache-apisix/apisix-dashboard/api/internal/log/zap.go:98 +0x208
github.com/apisix/manager-api/internal/log.GetLogger(0x0, 0x14)
	/Users/imjoey/Work/apache-apisix/apisix-dashboard/api/internal/log/zap.go:38 +0x3f
github.com/apisix/manager-api/internal/log.InitLogger(...)
	/Users/imjoey/Work/apache-apisix/apisix-dashboard/api/internal/log/zap.go:35
github.com/apisix/manager-api/internal/log.init.0()
	/Users/imjoey/Work/apache-apisix/apisix-dashboard/api/internal/log/zap.go:32 +0x26
```

The reason of this log error is the same as `conf.WorkDir` . So I commented out https://github.com/apache/apisix-dashboard/blob/62d1c439f37c98ca459f1ce2a0c15273575d2bce/api/internal/log/zap.go#L31-L33 .

Then I added `log.InitLogger()` after the line of code: https://github.com/apache/apisix-dashboard/blob/62d1c439f37c98ca459f1ce2a0c15273575d2bce/api/cmd/managerapi.go#L59 .

After done all the above, the bugs gone and everything worked fine. While honestly speaking, I'm not quite sure about the relationship between the **integration tests**and those two `init()` functions. You will be much appreciated if anyone can elaborate on it? Thanks.

--- a/Makefile
+++ b/Makefile
@@ -58,7 +58,7 @@ endif
 ### api-test:		Run the tests of manager-api
 .PHONY: api-test
 api-test: api-default
-	cd api/ && APISIX_API_WORKDIR=$$PWD go test -v -race -cover -coverprofile=coverage.txt -covermode=atomic ./...
+	cd api/ && APISIX_API_WORKDIR=$$PWD ENV=test go test -v -race -cover -coverprofile=coverage.txt -covermode=atomic ./...
 
 
 ### api-run:		Run the manager-api

--- a/api/cmd/managerapi.go
+++ b/api/cmd/managerapi.go
@@ -57,6 +57,7 @@ func NewManagerAPICommand() *cobra.Command {
 		Short: "APISIX Manager API",
 		RunE: func(cmd *cobra.Command, args []string) error {
 			conf.InitConf()
+			log.InitLogger()
 			droplet.Option.Orchestrator = func(mws []droplet.Middleware) []droplet.Middleware {
 				var newMws []droplet.Middleware
 				// default middleware order: resp_reshape, auto_input, traffic_log

--- a/api/internal/conf/conf.go
+++ b/api/internal/conf/conf.go
@@ -100,11 +100,6 @@ type Config struct {
 	Authentication Authentication
 }
 
-// TODO: it is just for integration tests, we should call "InitLog" explicitly when remove all handler's integration tests
-func init() {
-	InitConf()
-}
-
 func InitConf() {
 	//go test
 	if workDir := os.Getenv("APISIX_API_WORKDIR"); workDir != "" {

--- a/api/internal/conf/conf.go
+++ b/api/internal/conf/conf.go
@@ -34,6 +34,7 @@ const (
 	EnvBETA  = "beta"
 	EnvDEV   = "dev"
 	EnvLOCAL = "local"
+	EnvTEST  = "test"
 
 	WebDir = "html/"
 )
@@ -98,6 +99,13 @@ type Authentication struct {
 type Config struct {
 	Conf           Conf
 	Authentication Authentication
+}
+
+// ENV=test is for integration tests only, other ENV should call "InitConf" explicitly
+func init() {
+	if env := os.Getenv("ENV"); env == EnvTEST {
+		InitConf()
+	}
 }
 
 func InitConf() {

--- a/api/internal/conf/conf.go
+++ b/api/internal/conf/conf.go
@@ -101,6 +101,7 @@ type Config struct {
 	Authentication Authentication
 }
 
+// TODO: we should no longer use init() function after remove all handler's integration tests
 // ENV=test is for integration tests only, other ENV should call "InitConf" explicitly
 func init() {
 	if env := os.Getenv("ENV"); env == EnvTEST {

--- a/api/internal/log/zap.go
+++ b/api/internal/log/zap.go
@@ -27,6 +27,7 @@ import (
 
 var logger *zap.SugaredLogger
 
+// TODO: we should no longer use init() function after remove all handler's integration tests
 // ENV=test is for integration tests only, other ENV should call "InitLogger" explicitly
 func init() {
 	if env := os.Getenv("ENV"); env == conf.EnvTEST {

--- a/api/internal/log/zap.go
+++ b/api/internal/log/zap.go
@@ -27,9 +27,17 @@ import (
 
 var logger *zap.SugaredLogger
 
+// ENV=test is for integration tests only, other ENV should call "InitLogger" explicitly
+func init() {
+	if env := os.Getenv("ENV"); env == conf.EnvTEST {
+		InitLogger()
+	}
+}
+
 func InitLogger() {
 	logger = GetLogger(ErrorLog)
 }
+
 func GetLogger(logType Type) *zap.SugaredLogger {
 	writeSyncer := fileWriter(logType)
 	encoder := getEncoder(logType)

--- a/api/internal/log/zap.go
+++ b/api/internal/log/zap.go
@@ -27,10 +27,6 @@ import (
 
 var logger *zap.SugaredLogger
 
-// TODO: it is just for integration tests, we should call "InitLog" explicitly when remove all handler's integration tests
-func init() {
-	InitLogger()
-}
 func InitLogger() {
 	logger = GetLogger(ErrorLog)
 }

--- a/api/test/shell/cli_test.sh
+++ b/api/test/shell/cli_test.sh
@@ -126,6 +126,28 @@ if [[ $res != "hi~" ]]; then
 fi
 clean_up
 
+# run with -p flag out of the default directory
+workDir=$(pwd)
+rm -fr bin && mkdir bin
+cp ./manager-api ./bin/
+rm -rf html && mkdir html
+cd html
+echo "hi~" >> index.html
+$workDir/bin/manager-api -p $workDir &
+sleep 5
+
+res=$(curl http://127.0.0.1:9000)
+pkill -f manager-api
+cd $workDir
+rm -rf bin
+rm -rf html
+
+if [[ $res != "hi~" ]]; then
+    echo "failed: manager-api can't run with -p flag out of the default directory"
+    exit 1
+fi
+clean_up
+
 # test start info
 
 LOGLEVEL=$(cat conf/conf.yaml | awk '$1=="level:"{print $2}')

--- a/api/test/shell/cli_test.sh
+++ b/api/test/shell/cli_test.sh
@@ -128,24 +128,23 @@ clean_up
 
 # run with -p flag out of the default directory
 workDir=$(pwd)
-rm -fr bin && mkdir bin
-cp ./manager-api ./bin/
-rm -rf html && mkdir html
-cd html
-echo "hi~" >> index.html
-$workDir/bin/manager-api -p $workDir &
+distDir=/tmp/manager-api
+cp -r $workDir $distDir
+cd $distDir
+rm -fr bin && mkdir bin && mv ./manager-api ./bin/
+rm -rf html && mkdir html && echo "hi~" >> html/index.html
+cd bin && ./manager-api -p $distDir &
 sleep 5
 
 res=$(curl http://127.0.0.1:9000)
 pkill -f manager-api
-cd $workDir
-rm -rf bin
-rm -rf html
+rm -fr $distDir
 
 if [[ $res != "hi~" ]]; then
     echo "failed: manager-api can't run with -p flag out of the default directory"
     exit 1
 fi
+cd $workDir
 clean_up
 
 # test start info

